mkt/langpacks/utils.py

```diff
@@ -0,0 +1,43 @@
+ # -*- coding: utf-8 -*-
+from django import forms
+
+from tower import ugettext as _
+
+from mkt.files.utils import WebAppParser
+
+
+# FIXME: needs test, by mocking get_json_data() in various situations.
+class LanguagePackParser(WebAppParser):
+    def parse(self, upload):
+        """Parse the FileUpload passed in argument and return langpack data.
+        May raise forms.ValidationError()"""
+        # At the moment, we don't care about most of the manifest for
+        # langpacks so we simply extract the minimum we require ; in the future
+        # we might want to call the parent's parse() method and have a way to
+        # deal with extra stuff in the child.
+        data = self.get_json_data(upload)
+
+        languages_provided = data.get('languages-provided', {}).keys()
+        if len(languages_provided) < 1:
+            raise forms.ValidationError(_(u'Your language pack must contain '
+                    'one language in `languages-provided` object'))
+        if len(languages_provided) > 1:
+            # For now, refuse langpacks containing more than one language.
+            raise forms.ValidationError(_(u'Your language pack contains too '
+                    'many languages. Only one language per pack is supported'))
+
+        languages_target = data.get('languages-target', {}).values()
+        if len(languages_target) < 1:
+            raise forms.ValidationError(_(u'Your language pack must contain '
+                    'one language in `languages-target` object'))
+        if len(languages_target) > 1:
+            # For now, refuse langpacks containing more than one target.
+            raise forms.ValidationError(_(u'Your language pack contains too '
+                    'many targets. Only one target per pack is supported'))
```

Comment:
I think I'm going to add a check that the target is exactly `"app://*.gaiamobile.org/manifest.webapp"` and refuse anything else. And not care about what's in `"languages-provided"` beside the language itself.